core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -54,6 +58,22 @@ object MessageSet {
    */
   def entrySize(message: Message): Int = LogOverhead + message.size
 
+  /**
+   * Validate the "magic" values of messages are the same in a compressed message set and return the max timestamp
+   * of the inner messages.
+   */
+  def validateMagicValuesAndGetTimestamp(messages: Seq[Message]): Long = {
+    val sampleMagicValue = messages.head.magic
+    var largestTimestamp: Long = Message.NoTimestamp
+    for (message <- messages) {
+      if (message.magic != sampleMagicValue)
+        throw new IllegalStateException("Messages in the same compressed message set must have same magic value")
```

Comment:
The message in the exception can be mis-leading since validateMagicValuesAndGetTimestamp may not be called on a set of uncompressed messages.